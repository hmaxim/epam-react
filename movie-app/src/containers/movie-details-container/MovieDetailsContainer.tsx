import React, { useEffect } from 'react';
import MovieDetailsWrapper from './MovieDetailsWrapper';
import { connect } from 'react-redux';
import {
  getMovieById,
  loadMovies,
  setSearchParams,
} from '../../redux/rootActions';
import { useLocation, useParams } from 'react-router';
import EmptyState from '../../shared-components/empty-state/EmptyState';

const MovieDetailsContainer = (props: any) => {
  const location = useLocation();
  const params: any = useParams();

  useEffect(() => {
    if (props.history.location.pathname.includes('film')) {
      props.getMovieById(params.id);
      props.setSearchParams({ searchBy: 'genre' });
      props.loadMovies();
    }
  }, [location]);

  const genres = (genres: string[]) => <span>{genres.join(', ')}</span>;

  return props.selectedMovie && Object.keys(props.selectedMovie).length ? (
    <MovieDetailsWrapper score={props.selectedMovie.vote_average}>
      <div className="movie-poster">
        <img src={props.selectedMovie.poster_path} alt="poster" />
      </div>
      <div className="movie-details">
        <div className="movie-title-container">
          <h1 className="title">{props.selectedMovie.title}</h1>
          <div className="score">
            {+props.selectedMovie.vote_average.toFixed(1)}
          </div>
        </div>
        <div className="genre">{genres(props.selectedMovie.genres)}</div>
        <div className="year-time">
          <span>{props.selectedMovie.release_date} year</span>
          <span>{props.selectedMovie.runtime} min</span>
        </div>
        <div className="description">
          <span>{props.selectedMovie.overview}</span>
        </div>
      </div>
    </MovieDetailsWrapper>
  ) : <EmptyState>Film not found</EmptyState>;
};

const mapStateToProps = (state: any) => {
  return {
    movies: state.movies,
    loading: state.loading,
    error: state.error,
    selectedMovie: state.selectedMovie,
  };
};
const mapDispatchToProps = {
  loadMovies,
  getMovieById,
  setSearchParams,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MovieDetailsContainer);
