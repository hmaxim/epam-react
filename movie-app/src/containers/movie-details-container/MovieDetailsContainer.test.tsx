import 'jsdom-global/register';
import React from 'react';
import { render, shallow, mount } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import toJson from "enzyme-to-json";
import MovieDetailsContainer from './MovieDetailsContainer';
import { IMovie } from '../../interfaces/IMovie';

const MOVIE: IMovie = {
  id: 0,
  title: 'Titanic',
  tagline: null,
  vote_average: 2,
  vote_count: 0,
  release_date: '1997',
  poster_path: '../../assets/images/netflix-streaming-vs-traditional-cable.jpg',
  overview:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
  budget: 0,
  revenue: 0,
  runtime: 150,
  genres: ['Action'],
};

describe('ListHeaderContainer component', () => {
  let component: any;

  beforeEach(() => {
    component = render(<MovieDetailsContainer {...MOVIE} />);
  });

  test('should be rendered', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  /**
   * uncommit when jest-styled-components will be fixed
   */
//   test('should define props in MovieDetailsWrapper', () => {
//       expect(component.find('.score')).toHaveStyleRule('color', 'orange'); 
//   });
});
