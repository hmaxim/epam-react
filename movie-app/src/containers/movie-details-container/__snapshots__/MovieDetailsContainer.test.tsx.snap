// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeaderContainer component should be rendered 1`] = `
.c0 {
  height: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0 9.5%;
}

.c0 .movie-poster {
  width: 323px;
}

.c0 .movie-details {
  margin-left: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: calc(100% - 375px);
}

.c0 .movie-details .movie-title-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .movie-details .movie-title-container .title {
  color: white;
  display: inline-block;
  margin-right: 40px;
}

.c0 .movie-details .movie-title-container .score {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: orange;
  border: 1px solid white;
  border-radius: 50%;
  width: 50px;
  height: 50px;
}

.c0 .movie-details .genre {
  color: white;
}

.c0 .movie-details .year-time {
  color: #f65261;
}

.c0 .movie-details .year-time span {
  display: inline-block;
  margin-right: 40px;
}

.c0 .movie-details .description {
  margin-top: 50px;
  color: white;
  font-size: 14px;
}

<div
  class="c0"
>
  <div
    class="movie-poster"
  >
    <img
      alt="poster"
      src="[object Object]"
    />
  </div>
  <div
    class="movie-details"
  >
    <div
      class="movie-title-container"
    >
      <h1
        class="title"
      >
        Titanic
      </h1>
      <div
        class="score"
      >
        2.0
      </div>
    </div>
    <div
      class="genre"
    >
      <span>
        Action
      </span>
    </div>
    <div
      class="year-time"
    >
      <span>
        1997 year
      </span>
      <span>
        150 min
      </span>
    </div>
    <div
      class="description"
    >
      <span>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      </span>
    </div>
  </div>
</div>
`;
